app/controllers/users_controller.rb

```diff
@@ -20,9 +20,23 @@ def destroy
     redirect_to users_url
   end
 
-  def create
-    @user = User.new user_params
-    attempt_save
+  def import
+    if request.post?
+      xml = Nokogiri::XML params.require(:file).open
+      statuses = User.import_from_xml(xml)
+      if statuses
+        message = "Imported #{statuses[:imported]} new users"
+        unless statuses[:updated].zero?
+          message += " and updated #{statuses[:updated]}"
+        end
+        message += '.'
+        unless statuses[:rejected].zero?
+          message << " #{statuses[:rejected]} were rejected."
+        end
+        redirect_to users_url, notice: message
```

Comment:
That's what I was thinking of. I'm not sure about the specifics, but I would think any rejection would cause for an alert.